src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/helpers/HelmHelper.ps1

```diff
@@ -299,3 +299,121 @@ function Invoke-ExternalCommand {
+        return
+    }
+
+    if (Test-Path Env:HELMREGISTRY) {
+        $RegistryPath = Get-ChildItem -Path Env:HELMREGISTRY
+    }
+    else {
+        $ReleaseTrain = ''
+        if ((Test-Path Env:RELEASETRAIN) -and (Test-Path Env:RELEASETRAIN)) {
+            $ReleaseTrain = Get-ChildItem -Path Env:RELEASETRAIN
+        }
+        else {
+            $ReleaseTrain = 'stable'
+        }
+        $AzLocation = Get-AzLocation | Where-Object { ($_.DisplayName -ieq $Location) -or ($_.Location -ieq $Location) }
+        $Region = $AzLocation.Location
+        if ($null -eq $Region) {
+            Write-Error "Invalid location: $Location"
+            return
+        }
+        else {
+            $Location = $Region
+        }
+        $ChartLocationUrl = "https://${Location}.dp.kubernetesconfiguration.azure.com/azure-arc-k8sagents/GetLatestHelmPackagePath?api-version=2019-11-01-preview&releaseTrain=${ReleaseTrain}"
+
+        $Uri = [System.Uri]::New($ChartLocationUrl)
+        $Account = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureRmProfileProvider]::Instance.Profile.DefaultContext.Account
+        $Env = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureEnvironment]::PublicEnvironments[[Microsoft.Azure.Commands.Common.Authentication.Abstractions.EnvironmentName]::AzureCloud]
+        $TenantId = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureRmProfileProvider]::Instance.Profile.DefaultContext.Tenant.Id
+        $PromptBehavior = [Microsoft.Azure.Commands.Common.Authentication.ShowDialog]::Never
+        $Token = [Microsoft.Azure.Commands.Common.Authentication.AzureSession]::Instance.AuthenticationFactory.Authenticate($account, $env, $tenantId, $null, $promptBehavior, $null)
+        $AccessToken = $Token.AccessToken
+
+        $HeaderParameter = @{
+            "Authorization" = "Bearer $AccessToken"
+        }
+
+        $Response = Invoke-WebRequest -Uri $Uri -Headers $HeaderParameter -Method Post -UseBasicParsing
+        if ($Response.StatusCode -eq 200) {
+            $RegistryPath = ($Response.Content | ConvertFrom-Json).repositoryPath
+        }
+        else {
+            Write-Error "Error while fetching helm chart registry path: ${$Response.RawContent}"
+            return
+        }
+    }
+    Set-Item -Path Env:HELM_EXPERIMENTAL_OCI -Value 1
+    return $RegistryPath
+}
+
+function Get-HelmReleaseNamespaces {
+    param (
+        [string]$KubeConfig,
+        [string]$KubeContext
+    )
+    Write-Debug "Getting release namespace."
+    $ReleaseInstallNamespace = Get-ReleaseInstallNamespace
+    $ReleaseNamespace = $null
+    try {
+        $ReleaseNamespace = (helm status azure-arc -o json --kubeconfig $KubeConfig --kube-context $KubeContext -n $ReleaseInstallNamespace 2> $null | ConvertFrom-Json).namespace
+    }
+    catch {
+        Write-Error "Fail to find the namespace for azure-arc."
+    }
+    return , @($ReleaseNamespace, $ReleaseInstallNamespace)
```

Comment:
I wrote this but I now think returning a hash is nicer as it emphasises what we are returning.

Reply:
Cool, I'll make it to return a hash table.